pkg/client/fake/client.go

```diff
@@ -1278,3 +1315,88 @@ func zero(x interface{}) {
 	res := reflect.ValueOf(x).Elem()
 	res.Set(reflect.Zero(res.Type()))
 }
+
+func extractScale(obj client.Object) (autoscalingv1.Scale, error) {
+	switch obj := obj.(type) {
+	case *appsv1.Deployment:
```

Comment:
Is go smart enough to allow expression this in a single switch case:
```suggestion
	case *appsv1.Deployment, *appsv1.ReplicaSet:
```

Reply:
No, sadly :/